lib/base-compiler.js

```diff
@@ -1139,6 +1139,10 @@ export class BaseCompiler {
         }
 
         const inputFilenameSafe = this.filename(inputFilename);
+        
+        if (this.getOutputFilebase) {
+            this.outputFilebase = this.getOutputFilebase(inputFilename);
```

Comment:
This is not a good idea as `this` is shared among all compilations for 1 compiler. You can assume your `outputFilebase` will be `output` for now.

Reply:
Hi, I've tried to work around this - I've had to change from running the compiler to compiling the program from within the Erlang REPL to be able to set an output file other than `${inputFilename}.S`.

I assume there's a better way to do this though & don't believe the solution in the latest commit is really that great.